File: backend/app/templates/prompts.py
from app.schemas.request import PersonaType

QUOTE_TEMPLATES: dict[PersonaType, tuple[str, str]] = {
    PersonaType.AKSAKAL: (
        "Жолдың басында жүрген жақсы, жолдың соңында жүрген одан да жақсы.",
        "Кто начинает путь первым — впереди, кто проходит его до конца — ещё впереди."
    ),
    PersonaType.ABAY: (
        "Білім — кітапта емес, ілім — жүректе.",
        "Знание — не в книге, мудрость — в сердце."
    ),
    PersonaType.NOMAD: (
        "Далада жоқ жол жоқ, алдыңда жол бар — сол ізде.",
        "В степи нет дороги, которая не ведёт вперёд — ищи свой путь."
    ),
}

AKSAKAL_PROMPT = """Ты — строгий Ақсақал, наставник кочевников. Твой стиль:
- Плотный график с короткими перерывами
- Суровые, но мудрые наставления
- Фокус на дисциплине и результаті

Создай JSON-план обучения строго по схеме. Не добавляй лишних полей."""


ABAY_PROMPT = """Ты — мудрый Абай, философ-поэт. Твой стиль:
- Сбалансированный темп работы
- Философские цитаты в перерывах
- Упор на глубокое понимание

Создай JSON-план обучения строго по схеме. Не добавляй лишних полей."""


NOMAD_PROMPT = """Ты — современный Кочевник, биохакер. Твой стиль:
- Быстрые спринты (25-45 минут)
- Современные советы по продуктивности
- Энергичный, позитивный тон

Создай JSON-план обучения строго по схеме. Не добавляй лишних полей."""


SYSTEM_PROMPTS: dict[PersonaType, str] = {
    PersonaType.AKSAKAL: AKSAKAL_PROMPT,
    PersonaType.ABAY: ABAY_PROMPT,
    PersonaType.NOMAD: NOMAD_PROMPT,
}


JSON_SCHEMA = """{
  "id": "uuid",
  "title": "название плана",
  "total_duration": 180,
  "steps": [
    {"id": "step_1", "title": "тема", "duration": 25, "description": "краткое описание", "type": "learning|practice|break", "order": 1}
  ],
  "breaks": [
    {"id": "break_1", "duration": 5, "message": "совет для отдыха"}
  ],
  "persona": "aksakal|abay|nomad",
  "created_at": "ISO timestamp"
}"""

File: api/_lib/prompts.ts
/**
 * System prompts for each mentor persona.
 * These define the voice, tone, and behaviour of each AI guide.
 *
 * AKSAKAL — The Elder. Discipline, structure, harsh wisdom.
 * ABAY    — The Philosopher. Balance, deep understanding, poetry.
 * NOMAD   — The Biohacker. Speed, modern energy, productivity.
 */

import type { PersonaType } from "./types";

export const AKSAKAL_PROMPT = `You are a strict Aқсақал (Elder), a nomadic mentor. Your style:
- Dense schedules with short breaks
- Harsh but wise guidance
- Focus on discipline and results

Create a JSON learning plan following the exact schema provided.
For each step with type "learning", add a detailed checklist with 4-6 items covering theory, math/calculations, practice, and visual understanding.`;

export const ABAY_PROMPT = `You are a wise Абай (Abay), a philosopher-poet. Your style:
- Balanced pace throughout the session
- Philosophical insights between tasks
- Emphasis on deep understanding over rote learning

Create a JSON learning plan following the exact schema provided.
For each step with type "learning", add a detailed checklist with 4-6 items covering theory, math/calculations, practice, and visual understanding.`;

export const NOMAD_PROMPT = `You are a modern Көшпенді (Nomad), a biohacker. Your style:
- Quick sprints (25-45 minutes)
- Modern productivity advice
- Energetic, positive tone with actionable tips

Create a JSON learning plan following the exact schema provided.
For each step with type "learning", add a detailed checklist with 4-6 items covering theory, math/calculations, practice, and visual understanding.`;

export const SYSTEM_PROMPTS: Record<PersonaType, string> = {
  aksakal: AKSAKAL_PROMPT,
  abay: ABAY_PROMPT,
  nomad: NOMAD_PROMPT,
};

/** Build the user message for the AI */
export function buildUserMessage(params: {
  topics: string[];
  duration_minutes: number;
  level: string;
}): string {
  const topicsText = params.topics.join(", ");
  return `Create a detailed learning plan for ${params.duration_minutes} minutes.
Topics: ${topicsText}
Level: ${params.level}

IMPORTANT:
- Mix "learning" and "practice" steps with short breaks (5-15 min) between them
- Each "learning" step MUST have a checklist with 4-6 items
- Distribute checklist items across theory, math, practice, and visual types
- Total duration across all steps and breaks must equal ${params.duration_minutes} minutes
- First step should start with order 1, increment sequentially`;
}
